src/change_trust_asset.js

```diff
@@ -0,0 +1,271 @@
+          assetA: Asset.fromOperation(liquidityPoolParameters.assetA()),
+          assetB: Asset.fromOperation(liquidityPoolParameters.assetB()),
+          fee: liquidityPoolParameters.fee()
+        });
+      default:
+        throw new Error(`Invalid asset type: ${ctAssetXdr.switch().name}`);
+    }
+  }
+
+  /**
+   * Returns the XDR object for this change trust asset.
+   * @returns {xdr.ChangeTrustAsset} XDR ChangeTrustAsset object.
+   */
+  toXDRObject() {
+    if (this.isNative()) {
+      return xdr.ChangeTrustAsset.assetTypeNative();
+    }
+
+    if (this.isLiquidityPool()) {
+      const { assetA, assetB, fee } = this.getLiquidityPoolParameters();
+      const lpConstantProductParamsXdr = new xdr.LiquidityPoolConstantProductParameters(
+        {
+          assetA: assetA.toXDRObject(),
+          assetB: assetB.toXDRObject(),
+          fee
+        }
+      );
+      const lpParamsXdr = new xdr.LiquidityPoolParameters(
+        'liquidityPoolConstantProduct',
+        lpConstantProductParamsXdr
+      );
+      return new xdr.ChangeTrustAsset('assetTypePoolShare', lpParamsXdr);
+    }
+
+    // pad code with null bytes if necessary
+    const padLength = this.code.length <= 4 ? 4 : 12;
+    const paddedCode = padEnd(this.code, padLength, '\0');
+
+    if (this.code.length <= 4) {
+      const xdrAsset = new xdr.AlphaNum4({
+        assetCode: paddedCode,
+        issuer: Keypair.fromPublicKey(this.issuer).xdrAccountId()
+      });
+      return new xdr.ChangeTrustAsset('assetTypeCreditAlphanum4', xdrAsset);
+    }
+
+    const xdrAsset = new xdr.AlphaNum12({
+      assetCode: paddedCode,
+      issuer: Keypair.fromPublicKey(this.issuer).xdrAccountId()
+    });
+    return new xdr.ChangeTrustAsset('assetTypeCreditAlphanum12', xdrAsset);
+  }
+
+  /**
+   * @returns {string} Asset code.
+   */
+  getCode() {
+    return clone(this.code);
+  }
+
+  /**
+   * @returns {string} Asset issuer.
+   */
+  getIssuer() {
+    return clone(this.issuer);
+  }
+
+  /**
+   * @returns {LiquidityPoolParameters} Liquidity pool parameters.
+   */
+  getLiquidityPoolParameters() {
+    return clone(this.liquidityPoolParameters);
+  }
+
+  /**
+   * @see [Assets concept](https://www.stellar.org/developers/guides/concepts/assets.html)
+   * @returns {string} Asset type. Can be one of the following:
+   *
+   * * `native`
+   * * `credit_alphanum4`
+   * * `credit_alphanum12`
+   * * `liquidity_pool_shares`
+   */
+  getAssetType() {
+    if (this.isNative()) {
+      return 'native';
+    }
+    if (this.isLiquidityPool()) {
+      return 'liquidity_pool_shares';
+    }
+    if (this.code.length >= 1 && this.code.length <= 4) {
+      return 'credit_alphanum4';
+    }
+    if (this.code.length >= 5 && this.code.length <= 12) {
+      return 'credit_alphanum12';
+    }
+
+    return null;
+  }
+
+  /**
+   * @returns {boolean} `true` if this change trust asset object is the native asset.
+   */
+  isNative() {
+    return this.code && !this.issuer;
```

Comment:
Should we validate on `this.code.toLowerCase() === 'xlm'` here?

Reply:
Fixed in https://github.com/stellar/js-stellar-base/pull/454/commits/19e30f661a8ecdfe48bd548167630d2cccfd50e0.